docs/docker.md

```diff
@@ -0,0 +1,247 @@
+  that this is available only from your own machine rather than being exposed.
+
+  You can repeat this flag to forward additional ports similarly.  For example,
+  you can expose some of the [spark ports], e.g.,: `-p 127.0.0.1:4040:4040`.
+
+* **`-v ~/myfiles:/notebooks/myfiles`**
+
+  As described above, we're using `--rm` to remove the container when the run
+  exits.  This is usually fine since pulling out files from these containers can
+  be a little complicated.  Instead, we use this flag to map a directory from
+  your machine (the `~/myfiles` on the LHS) to a directory that is available
+  inside the running container.  Any modifications to this directory that are
+  done by the docker are performed directly on the actual directory.
+
+  Note that the local directory follows the local filename conventions, so on
+  Windows you'd use a Windows-looking path.  On Windows you also need to share
+  the drive you want to use in the [docker settings].
+
+  The path on the right side is used inside the container and it is therefore a
+  linux path.  The MMLSpark image runs jupyter in the `/notebooks` directory, so
+  it is a good place for making your files available conveniently.
+
+  This flag can be used more than once, to make several directories available in
+  the running container.  Both paths must be absolute, so if you want to specify
+  a path relatively, you can use something like `-v
+  $PWD/myfiles:/notebooks/myfiles`.
+
+  With such directory sharing in place, you can create/edit notebooks, and code
+  in notebooks can use the shared directory for additional data, for example:
+
+      data = spark.read.csv('myfiles/mydata.csv')
+      ...
+      model.write().overwrite().save('myfiles/myTrainedModel.mml')
+
+* **`microsoft/mmlspark:0.5`**
+
+  Finally, this specifies an explicit version tag for the image that we want to
+  run.
+
+
+## Running the container as a server
+
+An alternative to running the docker image interactively with `-it` is running
+it in a "detached" mode, as a server, using the `-d` (or `--detach`) flag.  An
+additional flag that is useful for this is `--name` that gives a convenient
+label to the running image:
+
+    docker run -d --name my-mmlspark ...flags... microsoft/mmlspark
+
+When running in this mode, you can use
+
+* `docker stop my-mmlspark`:  to stop the image
+
+* `docker start my-mmlspark`: to start it again
+
+* `docker logs my-mmlspark`:  to see the log output it produced
+
+* And you can use `docker exec -it my-mmlspark bash` to start a shell in the
+  context of the running server container.  This is roughly equivalent to
+  starting a terminal in the jupyter interface.  This can also be used with a
+  docker image running as described in previous sections, though it's best to
+  add the `--name` flag so you know what the running image is called (otherwise
+  docker assigns it a random name).
+
+
+## Running other Spark executables
```

Comment:
I've seen people ask to connect a bash shell to a running container so they can run `top`, `ps` etc. Is it worth documenting that?